planner/core/optimizer.go

```diff
@@ -83,13 +83,16 @@ func BuildLogicalPlan(ctx sessionctx.Context, node ast.Node, is infoschema.InfoS
 }
 
 // CheckPrivilege checks the privilege for a user.
-func CheckPrivilege(pm privilege.Manager, vs []visitInfo) bool {
+func CheckPrivilege(pm privilege.Manager, vs []visitInfo) error {
 	for _, v := range vs {
 		if !pm.RequestVerification(v.db, v.table, v.column, v.privilege) {
-			return false
+			if v.err == nil {
```

Comment:
Where will `v.err` be set in the future? in `RequestVerification`?

Reply:
`v.err` is set in `appendVisitInfo`, but for refactoring purposes, it is permitted to be set to `nil`, which means `ErrPrivilegeCheckFail`.  I could also set it to `ErrPrivilegeCheckFail` in `appendVisitInfo` (in fact I would prefer this), but there is code in `planner/core/planbuilder.go` which uses an `append` directly and not `appendVisitInfo`.  I suspect this is a historical legacy, in which case I can refactor that to use `appendVisitInfo`.